libs/platform/core/src/services/hydrate/default-hydrate.service.ts

```diff
@@ -0,0 +1,117 @@
+import { inject } from '@spryker-oryx/di';
+import {
+  hydratableAttribute,
+  HYDRATE_ON_DEMAND,
+  PatchableLitElement,
+  rootInjectable,
+} from '@spryker-oryx/utilities';
+import { isServer, LitElement } from 'lit';
+import { AppRef, ComponentsPlugin } from '../../orchestration';
+import { HydrateService } from './hydrate.service';
+
+export const locationHydrationEvent = 'oryx.location';
+export const hydrateAllEvent = 'oryx.force-all';
+
+export class DefaultHydrateService implements HydrateService {
```

Comment:
I think core hydration logic, like initHydrateHooks, should be implemented and should work independently from the injector and provided services.

Also, if we are using service for initiating full rehydrate, why can't we just inject service an call some method?